taca/cli.py

```diff
@@ -0,0 +1,32 @@
+import os
+
+import click
+import taca.log
+
+from pkg_resources import iter_entry_points
+
+from taca.utils import config
+
+
+@click.group()
+# Priority for the configuration file is: environment variable > -c option > default
+@click.option('-c', '--config-file',
+			  default=os.path.join(os.environ['HOME'], '.taca/taca.yaml'),
+			  envvar='TACA_CONFIG',
+			  type=click.File('r'),
+			  help='Path to TACA configuration file')
+@click.pass_context
+def cli(ctx, config_file):
+	""" Tool for the Automation of Storage and Analyses """
+
+	ctx.obj = {}
+	ctx.obj['config'] = config.load_yaml_config(config_file)
+	log_file = ctx.obj['config'].get('log', {}).get('log_file', None)
+	if log_file:
+		level = ctx.obj['config'].get('log').get('log_level', 'INFO')
+		taca.log.init_logger_file(log_file, level)
+
+
+#Add subcommands dynamically to the CLI
+for entry_point in iter_entry_points('taca.subcommands'):
```

Comment:
For reference for others, [this](http://click.pocoo.org/3/commands/#custom-multi-commands) is the relevant piece of docs. 

@robinandeer isn't this more useful when you want to build a more complex plugin architecture? As we are doing it now (using entry point) it *seems* like this would only introduce a bit of complexity to the code. At the end of the day you're doing exactly the same in this class than on the previous for loop, isn't it? You still have to define the subcommands in `setup.py`.

Am wrong? what do you think? :smiley: